spec/requests/groups_controller_spec.rb

```diff
@@ -1584,6 +1584,157 @@ def expect_type_to_return_right_groups(type, expected_group_ids)
       end
     end
 
+    describe '#add_owners' do
+      context "when logged in as an admin" do
+        before  { sign_in(admin) }
+
+        it 'should work' do
+          put "/groups/#{group.id}/owners.json", params: {
+            usernames: [user.username, admin.username].join(",")
+          }
+
+          expect(response.status).to eq(200)
+
+          response_body = response.parsed_body
+
+          expect(response_body["usernames"]).to contain_exactly(user.username, admin.username)
+
+          expect(group.group_users.where(owner: true).map(&:user))
+            .to contain_exactly(user, admin)
+        end
+
+        it 'returns not-found error when there is no group' do
+          group.destroy!
+
+          put "/groups/#{group.id}/owners.json", params: {
+            usernames: user.username
+          }
+
+          expect(response.status).to eq(404)
+        end
+
+        it 'does not allow adding owners to an automatic group' do
+          group.update!(automatic: true)
+
+          expect do
+            put "/groups/#{group.id}/owners.json", params: {
+              usernames: user.username
+            }
+          end.to_not change { group.group_users.count }
```

Comment:
Im always suspicious of this kind of code without reload it might regress due to some callbacks and we wouldn't get it I think

Reply:
Reading this again, it wasn't very clear. What I meant: `group.group_users.reload.count` and `group.group_users.count` could have different output based on callbacks added to models

Reply:
Interesting take! I was always a bit suspicious about code that required `reload` to test, but I guess we're onto the same thing, just from opposite directions. 🙂 Do you think this should be reloaded in anticipation of future changes, or you think whoever adds a callback later should notice the test failing and address?